### PR TITLE
[CMake] Added basic support for MSVS 2019 and vcpkg

### PR DESCRIPTION
**Pull request by AYESDIE**
Steps:
- Run `.\1-install-vcpkg.bat` in the vcpkg directory using PowerShell.
- Open the project in VS 2019.
- Use `Visual Studio 16 2019 Win64` as generator. 
- Set `-DCMAKE_TOOLCHAIN_FILE=...` parameter.

### CODE REVIEW

**Comment by morevnaproject** [comment]
OMG! :trophy:

**Comment by AYESDIE** [comment]
It still needs more work as of setting up MLT and other dependencies on *vcpkg* itself. This just completes the CMake part of configuring the project without the dependencies that are not available in *vcpkg*. It won't build it successfully yet.

File: 1-install-vcpkg.bat [hunk 1]
```diff
@@ -0,0 +1 @@
+vcpkg.exe install boost:x64-windows fftw3:x64-windows gettext:x64-windows glibmm:x64-windows gtkmm:x64-windows libxmlpp:x64-windows libsigcpp:x64-windows pango:x64-windows sdl2-mixer:x64-windows zlib:x64-windows
```
Comment (ice0) [inline 1-install-vcpkg.bat:1]: I suggest to use `%VCPKG_DEFAULT_TRIPLET%` environment variable, so we can easily switch between them.
```
set VCPKG_DEFAULT_TRIPLET=x64-windows
c:/projects/vcpkg/vcpkg.exe install boost-odeint fftw3 gettext glibmm gtkmm libxmlpp libsigcpp pango sdl2-mixer zlib
```

Also we don't need the full BOOST installation, only Boost.odeint.

File: synfig-core/src/CMakeLists.txt [hunk 2]
```diff
@@ -23,25 +23,33 @@ if (CMAKE_USE_PTHREADS_INIT)
...
+    pkg_check_modules(MLT REQUIRED mlt++)
+    pkg_check_modules(FFTW REQUIRED fftw3)
+    pkg_check_modules(FT REQUIRED freetype2) # for lyr_freetype
+    pkg_check_modules(LIBPNG REQUIRED libpng) # for mod_png
+    #TODO(ice0): find solution for libmng
+    pkg_check_modules(LIBMNG libmng) # for mod_mng (set as optional as it is not correctly installed in Debian)
+    pkg_check_modules(LIBJPEG REQUIRED libjpeg) # for mod_mng
+    pkg_check_modules(OPENEXR OpenEXR) # for mod_openexr
+    pkg_check_modules(MAGICKCORE REQUIRED MagickCore) # for Magick++
+    pkg_check_modules(FONT_CONFIG FontConfig) # for FontConfig
```
Comment (ice0) [inline synfig-core/src/CMakeLists.txt:42]: fontconfig

**Comment by ice0** [comment]
Merged. Thank you!

**Pull request merged**